strapi/strapi
# wip: static preview draft & publish tabs

### What does it do?

Add Draft / Publish tabs to switch between draft and live content in Static Preview.

![image](https://github.com/user-attachments/assets/7a7e7373-3cf7-467f-b76c-9ca1b2c2dcbb)

Requirements:
- Clicking on draft tab will return the status=draft url from the config
- Clicking on published tab will return the status=published url from the config
- If document is not published, only the draft tab will be available
- If the content type doesn't have draft and publish, tabs will not be visible

### Why is it needed?

To preview both draft and published data.

### How to test it?
1. `.env` set environment variable
`STRAPI_FUTURE_PREVIEW=true`

2. (`src/features.js`) ensure preview feature flag is configured as:
```js
module.exports = ({ env }) => ({
  future: {
    preview: env.bool('STRAPI_FUTURE_PREVIEW', false),
  },
});
```

2. (`src/admin.js`) Update your preview config to return different urls for draft and published content, e.g:
```ts
  preview: {
    enabled: env.bool('PREVIEW_ENABLED', true),
    config: {
      handler: (uid, { documentId, locale, status }) => {
        if (status === 'published') {
          return 'https://docs.strapi.io/';
        }

        return 'https://docs.strapi.io/dev-docs/intro';
      },
    },
  },
```

3. (`config/middlewares.js`) update CSP policies:
```ts
module.exports = [
  // ...
  {
    name: 'strapi::security',
    config: {
      contentSecurityPolicy: {
        useDefaults: true,
        directives: {
          'frame-src': ["'self'", 'https://docs.strapi.io/'], // e.g render strapi.io docs
        },
      },
    },
  },
  // ...
];
```
